### PR TITLE
fix the login delay too long issue

### PR DESCRIPTION
**Pull request by paraselene**
## Description

I use threading.Lock for the delay. There will be only 1 account allowed to run the delay block at the same time.After that, it will sleep for args.login_delay.

## Motivation and Context

In the recent commit Beehive Multiple Account Support (#1367), the login delay changed to delay = args.accounts.index(account) * args.login_delay + ((random.random() - .5) / 2), there is no problem for the first time login.
However, if I enable account_search_interval, it will have huge problem.
If I am running for 100 accounts, And I need to swap for the account[0-99] to [100-199], the delay for account 100 will be 10mins(100 x 6s) and delay for account 200 will be 20min(199 x 6s). It will completely stop my map if I set account_search_interval.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by paraselene** [comment]
@FrostTheFox, @Janglinator, please check.

**Comment by sLoPPydrive** [comment]
Ha! I ran into this issue today and implemented a quick'n'dirty fix by calling `stagger_thread` only for the very first loop of the searcher thread. Your approach is nicer though. :D

**Comment by paraselene** [comment]
@FrostTheFox, Could you please check this PR, it should be merged as it has a bug right now when rotating accounts

**Comment by sclo012** [comment]
I think even if we swap in any new accounts, there is unnessary delay in the current develop branch, so it is a problem whenever it's after initialization.

**Comment by bluemode** [comment]
Will you rebase?

**Comment by Imoq** [comment]
Was pulling my hair off because of all the issues encountered when rotating accounts/one failed and then I found this... :) <3

**Comment by gabaod** [comment]
What do you need to rebase? Its a simple file edit to keep this alive..  This should have been merged 2-4 weeks ago.  Its sad that the devs are forcing PR devs to keep updating their stuff when this fixes bugs.

**Comment by sebastienvercammen** [comment]
@gabaod There is no "devs" vs "PR devs": PRs are open, and the PR rank on Discord is a rank open to everyone. All discussion about development is done in #pr and everyone involved has the option to point out necessary changes via Github's reviews.

If you have something constructive to add, add it. Otherwise keep irrelevant comments out so we can stay focused on moving forward.

**Comment by FrostTheFox** [comment]
@gabaod has been blocked from the GitHub, they have a history of needlessly insulting the project and it's developers despite reasoning and explanations being given for everything he mentions.

**Comment by bluemode** [comment]
Good job fox!

**Comment by FrostTheFox** [comment]
I'm not going to hide the fact that the flow of commits has definitely slowed down, but that's bound to happen when PoGo loses popularity in general, and devs leave the scene.

**Comment by catnet23** [comment]
@paraselene can you please fix the conflcts? would be nice.

File: pogom/search.py [hunk 1]
```diff
@@ -477,7 +479,12 @@ def search_worker_thread(args, account_queue, account_failures, search_items_que
             status['message'] = 'Switching to account {}.'.format(account['username'])
             log.info(status['message'])
 
-            stagger_thread(args, account)
+            # Delay each thread start time so that logins occur after delay.
```
Comment (onilton) [inline pogom/search.py]: Can this code be moved into `stagger_thread`? I mean, it was already beautifuly encapsulated....

**Comment by sebastienvercammen** [comment]
With the author gone, #1742 rebased and updated the code.

**Comment by paraselene** [comment]
@sebastienvercammen, Can't you see that I merge the develop branch and fix the conflict 6 days ago? why you say that I am gone?!

**Comment by sebastienvercammen** [comment]
@paraselene Just the messenger. I was told you were contacted a few times for separate PRs, but that there was no response.

**Comment by paraselene** [comment]
Since I am not in the #pr though I requested to join #pr in 23/12/2016, I don't know who are those #pr on Discord.

**Pull request closed without merging**